contracts/modules/RelayerManager.sol

```diff
@@ -291,7 +285,7 @@ abstract contract RelayerManager is BaseModule, SimpleOracle {
                 return false; // Signers must be different
             }
             lastSigner = signer;
-            (isGuardian, guardians) = Utils.isGuardianOrGuardianSigner(guardians, signer);
+            bool isGuardian = _isGuardianOrGuardianSigner(_wallet, signer);
```

Comment:
This makes it possible for a smart contract guardian to sign more than once by changing what `owner()` returns between subsequent calls. Note that this is not something we test for, which is why our tests do not fail.

Reply:
Actually, static call is not a protection. Here is an example of how a malicious smart contract guardian could still return multiple addresses:

```
pragma solidity ^0.8.1;

contract Guardian {
    function owner() external view returns (address) {
        uint idx = gasleft() % 10000;
        if(idx > 8000) return 0x5B38Da6a701c568545dCfcB03FcB875f56beddC4;
        if(idx > 4000) return 0xAb8483F64d9C6d1EcF9b849Ae677dD3315835cb2;
        return 0x4B20993Bc481177ec7E8f571ceCaE8A9e22C02db;
    }
}

contract Module {
    Guardian g;
    constructor() {
        g = new G();
    }
    function callGuardian() external view returns (address[] memory a) {
        a = new address[](3);
        for(uint i = 0; i < 3; i++) {
            a[i] = g.owner();
        }
    }
}
```